### PR TITLE
mmdblookup module cleanup

### PR DESCRIPTION
**Pull request by GrayTShirt**
this covers building, style, and readme

### CODE REVIEW

**Comment by rgerhards** [comment]
@GrayTShirt @PascalWithopf could you join forces to integrate into CI testing. I guess there is some overlap of work :-) @friedl we could also see if we can package this module.

@GrayTShirt it would also be great if you could add some doc at the rsyslog-doc project. Users usually do not even know about README files.

**Comment by GrayTShirt** [comment]
@rgerhards I can take care of the ci-building today.

would you like me to squash my commits before merging? So far I've kept them separate thinking that each commit was a distinct change worth tracking.

I may not be able to get to the documentation today.

**Comment by PascalWithopf** [comment]
@GrayTShirt could you add the module to the CONFIG_FLAGS in run.sh so travis is going to build it?

**Comment by GrayTShirt** [comment]
@PascalWithopf, no problem, but we're going to have to install libmaxminddb as part of the test process.

**Comment by GrayTShirt** [comment]
ugh, so many build problems with regards to libmaxminddb

**Comment by GrayTShirt** [comment]
@rgerhards there is documentation for mmdblookup already, https://github.com/rsyslog/rsyslog-doc/blob/master/source/configuration/modules/mmdblookup.rst

**Comment by rgerhards** [comment]
> @rgerhards there is documentation for mmdblookup already, https://github.com/rsyslog/rsyslog-doc/blob/master/source/configuration/modules/mmdblookup.rst

sorry for that ;-)

**Comment by rgerhards** [comment]
on CI testing: I think it is OK if we just make sure the module compiles. Real tests would of course be better, but if this is hard to get on CI, let's start with the easy bait.

File: tests/travis/install.sh [hunk 1]
```diff
@@ -8,6 +8,16 @@ sudo apt-get install -qq --force-yes libestr-dev librelp-dev libfastjson-dev lib
...
 sudo apt-get install -qq python-docutils
 
+if [ "$DISTRIB_CODENAME" == "trusty" ]; then
+	set -ex
+	WANT_MAXMIND=1.2.0
+	curl -Ls https://github.com/maxmind/libmaxminddb/releases/download/${WANT_MAXMIND}/libmaxminddb-${WANT_MAXMIND}.tar.gz | tar -xz
+	(cd libmaxminddb-${WANT_MAXMIND} ; ./configure --prefix=/usr CC=gcc CFLAGS="-Wall -Wextra -g -pipe -std=gnu99" ; sudo make install)
+	set +x
+else
+	sudo apt-get install -qq libmaxminddb-dev
```
Comment (rgerhards) [inline tests/travis/install.sh:18]: We don't need to test on all platforms. If it works on newer than trusty, we can simply use that and disable the tests on trusty. We do this with some other tests as well. It's also a question of CI runtime, which already is close to the limit. Usually, the functionality tests work well on all platforms if it works on one. Of course not always, but we need a compromise between CI coverage and runtime.

**Comment by rgerhards** [comment]
The commits look great to me as is - I definitly prefer not to squash them further.

**Comment by GrayTShirt** [comment]
I'm out of the office right now, when I get back I'll push the code changes to fit it to compile with clang. GitHub being down disrupted my workflow, got to take lunch though :)

**Comment by rgerhards** [comment]
No problem at all. There are still 5 weeks until the next release ;-) I am glad you are working on this.

**Comment by rgerhards** [comment]
@GrayTShirt I think we should probably strip this PR down to that CI checks the build process. The actual test can than be later provided via a different PR. That would enable us to merge the patch and solve the build issues. What do you think?

**Comment by GrayTShirt** [comment]
@rgerhards that seems like a good plan, progress is still progress. I'll go ahead and disable the functional tests, until I get the time to debug the module

**Comment by GrayTShirt** [comment]
so it looks like we failed an unrelated test.

**Comment by rgerhards** [comment]
indeed - looks like a timing issue on travis. Have restarted the test.

**Comment by GrayTShirt** [comment]
It looks like we failed another unrelated test.

**Comment by rgerhards** [comment]
@GrayTShirt mhh... that's a bit strange, these parts used to work quite reliably. But you are right, this is definitely unrelated, and if such random failures continue are an issue in themselves. I'll restart the build.

**Comment by GrayTShirt** [comment]
Hurray, tests passed.

**Comment by GrayTShirt** [comment]
@rgerhards consider this a friendly bump :)

**Comment by rgerhards** [comment]
No problem. Just FYI: depending on other work to do, I do not merge each day. However, I make sure that everything get's merged before the next release, usually at latest in the week before release.

**Comment by GrayTShirt** [comment]
no worries, I felt it was worth bothering you at least once a week.

**Pull request merged**